pkg/catalog/catalog.go

```diff
@@ -2229,10 +2257,18 @@ func (c *Catalog) RestoreRepositoryStatus(ctx context.Context, repositoryID stri
 	return &status, nil
 }
 
-// runBackgroundTaskSteps update task status provided after filling the 'Task' field and update for each step provided.
+// RunBackgroundTaskSteps update task status provided after filling the 'Task' field and update for each step provided.
 // the task status is updated after each step, and the task is marked as completed if the step is the last one.
-// initial update if the task is done before running the steps.
-func (c *Catalog) runBackgroundTaskSteps(repository *graveler.RepositoryRecord, taskID string, steps []taskStep, taskStatus protoreflect.ProtoMessage) error {
+// initial update if thRunBackgroundTaskStepse task is done before running the steps.
+func (c *Catalog) RunBackgroundTaskSteps(repository *graveler.RepositoryRecord, taskID string, steps []TaskStep, taskStatus protoreflect.ProtoMessage) error {
+	// Validate that APIErrorCB is properly configured
+	if c.APIErrorCB == nil {
+		return ErrAPIErrorCBNotSet
+	}
+	if c.APIErrorCB.GetHandlerType() != "controller.handleAPIErrorCallback" {
+		return fmt.Errorf("%w: expected 'controller.handleAPIErrorCallback', got '%s'", ErrAPIErrorCBHandlerTypeMismatch, c.APIErrorCB.GetHandlerType())
+	}
```

Comment:
Remove check. 
It breaks the abstraction of a callback, think of it this way - if we need a concrete type here than why do we acceppt a generic callback in the first place? 

Instead, make sure we initiate this properly.
This function sohuld be agnostic to what callback error was set and just call it.